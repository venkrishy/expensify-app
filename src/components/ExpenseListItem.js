import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
    return (
        <div>
            <p>
                <Link to={`/edit/${id}`}>{description}</Link> - ${amount} - {createdAt} 
            </p>
        </div>
    )
}

export default ExpenseListItem;