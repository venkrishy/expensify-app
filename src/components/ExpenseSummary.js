import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = ({ expenseCount, expenseTotal }) => {
    const expenseWord = expenseCount == 1 ? 'expense' : 'expenses';
    const totalText = numeral(expenseTotal / 100).format('$0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totaling <span>{totalText}</span></h1>
                <div className="page-header__actions" >
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};




const mapStateToProps = (state) => {
    return {
        expenseCount: selectExpenses(state.expenses, state.filters).length,
        expenseTotal: selectExpensesTotal(state.expenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);


