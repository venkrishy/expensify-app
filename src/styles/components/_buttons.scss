.button {
    background: #1c88bf;
    border: none;
    color: white;
    display: inline-block;
    border-radius: 3px;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
}


.button--secondary {
    background: #888888;
}