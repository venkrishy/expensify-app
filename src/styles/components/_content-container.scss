.content-container {
    margin: 0 auto;
    padding: 0 $m-size; /* None of our content pushes against the edge of the browser */
    max-width: 80rem;
    /* Inherits the background */
}