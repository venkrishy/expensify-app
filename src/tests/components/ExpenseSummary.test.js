import React from 'react';
import { shallow } from 'enzyme';
import {ExpenseSummary} from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

let editExpense, removeExpense, history, wrapper;

beforeEach(() => {
});

test("should render ExpenseSummary with one expense", () => {
    const wrapper = shallow(<ExpenseSummary expenses={[expenses[0]]} expenseTotal="9434" />);
    expect(wrapper).toMatchSnapshot();

});

test("should render ExpenseSummary with two expenses", () => {
    const wrapper = shallow(<ExpenseSummary expenses={expenses} expenseTotal="9434" />);
    expect(wrapper).toMatchSnapshot();

});
