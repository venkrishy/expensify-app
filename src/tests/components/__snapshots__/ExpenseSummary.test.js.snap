// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with one expense 1`] = `
<div>
  <p>
    Viewing 
     
    expenses
     totaling 
    $94.34
  </p>
</div>
`;

exports[`should render ExpenseSummary with two expenses 1`] = `
<div>
  <p>
    Viewing 
     
    expenses
     totaling 
    $94.34
  </p>
</div>
`;
